internal/controlplane/handlers_profile.go

```diff
@@ -96,6 +102,9 @@ func (s *Server) CreateProfile(ctx context.Context,
 		return nil, status.Errorf(codes.Internal, "error creating profile")
 	}
 
+	// Adds default rule names, if not present
+	populateRuleNames(in)
```

Comment:
It would make the logic simple but, the errors won't be as clear, consider this profile:
```yaml
repository:
  - type: dependabot_configured
    def:
      package_ecosystem: gomod
  - type: dependabot_configured
    def:
      package_ecosystem: npm
```
The user has not specified the name, but as we populate before validation, we would view the profile as:
```yaml
repository:
  - type: dependabot_configured
    name: dependabot_configured
    def:
      package_ecosystem: gomod
  - type: dependabot_configured
    name: dependabot_configured
    def:
      package_ecosystem: npm
```
And we would return some error like: `rule 'dependabot_configured' has the same name 'dependabot_configured' in entity 'repository'`, which the user might get confused by as they didn't specify any name.

Now, if you say, if `ruleName == ruleType` consider user supplied empty name. But that won't be true. Users can explicitly define the name to be equal to rule type, something like:
```yaml
repository:
  - type: dependabot_configured
    name: dependabot_configured
    def:
      package_ecosystem: gomod
  - type: dependabot_configured
    def:
      package_ecosystem: npm
```
This may sound very improbable, but folks actually do stuff like this while writing any yaml. 

---

For the original profile, an error message like `cannot have rules with empty name and same type in entity 'repository'` would be better and clearer to the user. Wdyt?

Reply:
I'm okay with either error message -- we _are_ changing the schema out from under users, so a message like

```
rule 'dependabot_configured' conflicts with 'dependabot_configured' in the same profile, assign a unique name to at least one.
```

would probably not confuse users even if they hadn't been using the `name` field previously.

Reply:
Done, I modified the wording of errors to give some sort of solution as well.